tools/SwiftSyntax/SerializedDiagnosticConsumer.swift

```diff
@@ -0,0 +1,296 @@
+//===---------- BitstreamWriter.swift - LLVM Bitstream Writer -------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+// This file provides an implementation of the Clang diagnostic serialization
+// format for diagnostics that have been added to a DiagnosticEngine.
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+class SerializedDiagnosticConsumer: DiagnosticConsumer {
+  enum BlockID: UInt8 {
+    // must match llvm::bitc::FIRST_APPLICATION_BLOCKID
+    case metadata = 8
+
+    case diagnostics
+
+    var name: String {
+      switch self {
+      case .metadata: return "Meta"
+      case .diagnostics: return "Diag"
+      }
+    }
+  }
+
+  enum RecordID: UInt8 {
+    case version = 1
+    case diagnostic
+    case sourceRange
+    case diagnosticFlag
+    case category
+    case filename
+    case fixIt
+
+    var name: String {
+      switch self {
+      case .version: return "Version"
+      case .diagnostic: return "DiagInfo"
+      case .sourceRange: return "SrcRange"
+      case .diagnosticFlag: return "DiagFlag"
+      case .category: return "CatName"
+      case .filename: return "FileName"
+      case .fixIt: return "FixIt"
+      }
+    }
+  }
+
+  let stream = BitstreamWriter()
```

Comment:
Do we need an application-specific bit stream writer in swiftc to actually emit these diags?